### DISCUSSION TITLE
Openid-connect/token endpoint taking more time

### DISCUSSION THREAD

**Discussion by kavaka123** [keycloak/keycloak#33885] [Q&A]
Hi Team,
Can someone please explain why the keycloak authentication endpoint is taking more time?(~120ms). The spans also shows the processes are completed quickly, but still the overall trace is quite high. We can also see there is some process delay after span4 and before span5. And also after the last span. Any suggestions/advise on how to reduce the overall api transaction time?

<img width="1639" alt="Screenshot 2024-10-14 at 2 40 06 PM" src="https://github.com/user-attachments/assets/dffeeccc-cfcf-41be-ae00-4791c5734727">

**Comment by codespearhead** [comment]
Please confirm that this issue is still reproducible in the latest stable release of Keycloak, which currently is version `26.0.0`, and that you're running Keycloak production mode (`kc.sh start` instead of `kc.sh start-dev`).

**Reply by kavaka123** [reply]
No, We are using **docker.io/bitnami/keycloak:23.0.7-debian-12-r4** version with production mode enabled. You want me to upgrade to the latest version and confirm?

**Reply by codespearhead** [reply]
Yes, you should ensure this issue is reproducible using the latest stable release of the official Keycloak OCI image [1][2][3][4]. If not, you should close this discussion [5] and reach out to that third-party OCI image's vendor for support, because whatever it does on top of the official one is out of the official project's control.

***

[1] https://github.com/keycloak/keycloak/discussions/29126
[2] https://github.com/keycloak/keycloak/discussions/29126#discussioncomment-9257666
[3] https://github.com/keycloak/keycloak/discussions/33407
[4] https://github.com/keycloak/keycloak/discussions/33407#discussioncomment-10820449
[5] https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#closing-a-discussion

**Reply by annejohnson** [reply]
@kavaka123 

Were you able to upgrade, and if so, did that fix the issue?

Or did you find another solution?

I'm seeing a similar issue in the same KeyCloak version (23.0.7).
